alphagov/paas-admin
# Add password reset functionality

What
----

We'll be sending out password reset email to our tenants. Notify
requires us to point out a template to be used for email sending
functionality.

We have a wrapper around notify's SDK, which is limited to our needs.
We're adding a new function that will specifically target the password
reset email.

We're able to use UAA headlessly via the API to perform some actions.
These include the password reset flow. We're simply adding functions
calling for these specific endpoints, with some additional data.

It seems that using both JSON.stringify and axios, messes up with the
attribute enough, for the UAA API to return bad request response.

This deserves testing the user invitation functionality...

As a bonus, I've had a look into the navigation component.

We're adding more and more pages which do not require authentication.
This makes it unreasonable, to ignore the navigation always say things
like `Sign out`. Instead, we're adding a flag, which will trigger
different menu items depending on the state of the session.

How to review
-------------

- Deploy alphagov/paas-cf#2328 to your dev env - or manually setup your env
- Deploy pazmin from this branch to your environment
- Attempt to run through entire forgotten password flow
- Check wording please - I made it all up...

⚠️ This needs to be merged before the alphagov/paas-cf#2328 ⚠️ 
-------------------------------------------------------------------
